core/src/block_sync.rs

```diff
@@ -244,6 +275,9 @@ pub mod message {
                     use crate::sumeragi::message::BlockSyncUpdate;
 
                     for block in blocks.clone() {
+                        block_sync
+                            .seen_blocks
+                            .insert((block.header().height().get(), block.hash()));
```

Comment:
what if we receive a block with the height that is already present in the `seen_blocks` but with a different hash?